src/main/java/com/asyncy/apis/hub/controllers/BaseController.java

```
@@ -1,36 +1,81 @@
     }
 
-    static HttpResponse<JsonNode> prepareGHRequest(final String path, final String oauthToken)
+    // the Accept header is needed to fetch the last version of the GitHub API with the topics inside
+    private static HttpResponse<JsonNode> prepareGHRequest(final String path, final String oauthToken)
             throws UnirestException {
         return Unirest.get("https://api.github.com" + path)
                 .header("Authorization", "bearer " + oauthToken)
+                .header("Accept", "application/vnd.github.mercy-preview+json")
                 .asJson();
     }
 
+    final HttpResponse<JsonNode> prepareGHRequest(final String path) throws UnirestException {
+        return prepareGHRequest(path, this.creds.getOauthToken());
+    }
+
+    final JsonNode makeGHRequest(final String path) throws UnirestException {
+        return prepareGHRequest(path).getBody();
+    }
+
     static JsonNode makeGHRequest(final String path, final String oauthToken) throws UnirestException {
         return prepareGHRequest(path, oauthToken).getBody();
     }
 
-    final String getOwnerUuid(final HttpServletRequest request) {
-        return getDecodedJWTToken(request)
-                .getClaim(Constants.JWT_CLAIM_KEY_OWNER_UUID).asString();
+    final User fetchUser(final HttpServletRequest request) {
+        try {
+            if (this.creds == null) {
+                this.creds = fetchOwner(request);
+            }
+            return new User(makeGHRequest("/user").getObject());
+        } catch (UnirestException e) {
+            return null;
+        }
+    }
+
+    final Repo getRepoByName(final String name) {
+        try {
+            HttpResponse<JsonNode> res = prepareGHRequest("/repos/" + name);
+            if (res.getStatus() == 200) {
+                return new Repo(res.getBody().getObject());
+            }
+            return null;
+        } catch (UnirestException | IOException e) {
+            return null;
```

Don't eat the exception, use log it using the logger.